### ISSUE TITLE
Request: detecting empty list in set.union(*[])

### ISSUE THREAD

**Issue by jamesbraza** [pylint-dev/pylint#8538]
### Current problem

Please see the below code snippet revolving around `set` construction:

```python
# Empty list
set.union(*[])  # classmethod: TypeError: unbound method set.union() needs an argument
set().union(*[])  # method: works

# Non-empty list of empty set
set.union(*[{}])  # classmethod: TypeError: descriptor 'union' for 'set' objects doesn't apply to a 'dict' object
set().union(*[{}])  # method: works

# Non-empty list of non-empty set
set.union(*[{1}])  # classmethod: works
set().union(*[{1}])  # method: works
```

`pylint=2.17.2` currently doesn't detect any of the known failure cases.

### Desired solution

It would be cool if `pylint` could detect the known failure cases.

### Additional context

This actually gets discussed in the comments of this Stack Overflow answer: https://stackoverflow.com/a/31253153
